android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java

```diff
@@ -0,0 +1,49 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.List;
+
+import io.realm.RealmResults;
+
+import static org.fossasia.openevent.utils.SortOrder.sortOrderSpeaker;
+
+public class SpeakersListFragmentViewModel extends ViewModel{
+
+    private MutableLiveData<List<Speaker>> speakersList;
+    private RealmDataRepository realmRepo;
+    private RealmResults<Speaker> realmResults;
+    private String searchText = "";
```

Comment:
Do we need this variable?

Reply:
Yes, i think we need to store the value of the Search text here after the activity is no longer active.

Reply:
But I can't see this variable being used anywhere while fetching the data inside ViewModel.
It only has a getter and setter.

Reply:
Sorry but i am getting slightly confused here. 
Here is the flow:
1. The activity gets created.(blank searchText variable is declared in the activity : line number 66)
2. The view model instance is created.(the blank searchText variable is created in the viewmodel : line number 21)
3. On line 85 in the activity we query the view model for the searchText.

Now let's say the screen is rotated, and the activity is recreated. Then again in the onCreate line number 85 , we will get the original search query back.

I can see the use of the variable here.

Reply:
@the-dagger Infact i think we needed that ```onSavedInstanceState``` method also. Since there is no way of setting the ```searchText``` string in the ViewModel.

Reply:
@srv-twry 
Looks like there is an issue with how the search behavior has been implemented here.
Instead of querying the data from the database for entered text, the existing data is being filtered.

>On line 85 in the activity we query the view model for the searchText.

Nope, take a look closely, the ViewModel always returns the entire list of speakers.

What you need to do here is, get rid of the query method in the adapter and implement the search the way it should be done.

You will need to create a new method inside your ViewModel that queries the database based on the string that was entered by the user and then returns the value back.

Reply:
Or maybe open a new issue here for properly implementing the search behavior.
The current implementation looks a bit hacky to me.

Reply:
@the-dagger Yes.
The ViewModel always returns the entire list of speakers and then the actual filtering is done inside the adapter using the "filter" method
That's how it was implemented before and has been implemented everywhere else.  Hence i thought i should be left untouched.

Reply:
No problem.
We can fix the implementation since we are anyways rewriting the Repository layer of the app.

Reply:
Infact i had a PR merged today which had the same issue. It was for LocationsFragment. There also the filter operation was done inside the adapter and hence i left it untouched.
Shall i open issues for each and every recyclerview because it's implemented in this way everywhere for eg in LocationsFragment, SpeakersListFragment, DayScheduleFragment, TracksFragment.
It's almost in each and every RecyclerView.

Reply:
Open an issue for this PR and after that with every ViewModel related PR, you can do this modification in that PR itself.